client/reader/discover/helper.jsx

```diff
@@ -17,11 +18,11 @@ module.exports = {
 	},
 
 	isDiscoverPost: function( post ) {
-		return !! post.discover_metadata;
+		return !! ( post.discover_metadata || post.site_ID == config( 'discover_blog_id' ) ); //eslint-disable-line eqeqeq
```

Comment:
site_ID is sometimes a Number, sometimes a string. Depends on where we get it from.